resources/fetchers/ecr_factory.go

```diff
@@ -0,0 +1,74 @@
+package fetchers
+
+import (
+	"fmt"
+	"github.com/docker/distribution/context"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/kubernetes"
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/cloudbeat/resources/manager"
+	"github.com/elastic/cloudbeat/resources/providers"
+	"github.com/elastic/cloudbeat/resources/providers/aws"
+	"regexp"
+)
+
+const (
+	ECRType = "aws-ecr"
+)
+
+func init() {
+	awsConfigProvider := aws.ConfigProvider{}
+	awsConfig := awsConfigProvider.GetConfig()
+	ecr := aws.NewEcrProvider(awsConfig.Config)
+	identityProvider := aws.NewAWSIdentityProvider(awsConfig.Config)
+	kubeGetter := providers.KubernetesProvider{}
```

Comment:
General note, I think we should try pushing all of these preparation lines of code into the `Create` method.
I realized it may mess up the tests though it makes more sense.
Since the test is on the same package as the factory, you can practically do whatever you want from within the test.

Reply:
So as we discussed I have made some research.

In general, go guidelines propose:
- No package-level variables
- No func init

Supporting articles 
https://leighmcculloch.com/posts/tool-go-check-no-globals-no-inits/
https://peter.bourgon.org/blog/2017/06/09/theory-of-modern-go.html

When diving deep a little bit deeper:
The only"approved" cases when init is legit are:
1. Registration of a package (Similar to the factory registration) 
2. Setup and initialization of specific package 
3. Variables initialization if cannot be done in initialization expr
4. running one-time computations

https://www.digitalocean.com/community/tutorials/understanding-init-in-go
https://medium.com/golangspec/init-functions-in-go-eac191b3860a
https://www.developer.com/languages/inti-function-golang/


Factory - 
```go 
func init() {
	Register("postgres", NewPostgreSQLDataStore)
	Register("memory", NewMemoryDataStore)
}
```
https://matthewbrown.io/2016/01/23/factory-pattern-in-golang

Reply:
So After my research, I have been wondering about few things:

1. What are the benefits of having `init()` function in each and every factory?
I guess that the obvious answer would be - because then I can customize each init (for each factory).
But then the question, why would we want to customize each factory?

We could for example - create only one init() function in the abstract factory and register all of the factories together. 
There is no mutual agreement on the right usage of the init(), but in general, the "godfathers" of the language suggest using it just in only extreme cases.

Implementing it this way can help us achieve more readability and reduce the usage of "magic" in our code.
BTW - we don't need to use init() at all and we can just put all of the register functions in the factory CTOR.
(Better explicit than implicit - Zen of python).

2. Should we put more logic in the init function?

So this is a very tricky question if looking at the first bullet - probably not because we shouldn't use init at all - or at least reduce that magic it does.

Implementation-wise, we will need to move the current init functionality to the "CreateForm" function.
But the thing that bothers me here - "CreateFrom" should be a private function. It feels a little bit smelly to test a private function (or at least a function that should be made private).

On the other hand, keeping the current solution where init() creates the factory is extending the "black magic" in our code, and will probably result in less readability.

Reply:
@amirbenun WDT?

Reply:
As we spoke with @yashtewari - changes were made